src/core/ir/model.rs

```diff
@@ -108,6 +109,14 @@ impl Cache {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct InputTransforms {
+    pub subfield_types: HashMap<(String, String), String>,
+    pub subfield_renames: HashMap<(String, String), String>,
+    pub arg_name: String,
+    pub arg_type: String,
+}
```

Comment:
I think this should be a recursive structure, something like:

```rust
pub enum Transform {
    Object(HashMap<FieldName, Transform>),
    Rename {from: String, to: String}
}
```

Reply:
I refactored it to document it better with code. There is no need to introduce a complex recursive type because I only go up to 1 level deep. That's why it is a (String, String) (type name, field name) => new field name. I use a hashmap for performance gains to check the presence of a rename for a field.